### PR TITLE
Add insertAt function

### PR DESCRIPTION
**Pull request by MartinSStewart**
Thanks @zwilias for the original code.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,2 +1,5 @@
 # Ignore build or dist files
 /elm-stuff
+
+# Ignore IntelliJ config files
```
Comment (jonboiser) [inline .gitignore]: I think you can remove this from the patch.

Comment (MartinSStewart) [reply .gitignore]: Fixed

File: src/Array/Extra.elm [hunk 2]
```diff
@@ -312,3 +313,30 @@ removeAt index xs =
...
+
+-}
+insertAt : Int -> a -> Array a -> Array a
+insertAt index val values =
+    let
+        length =
+            Array.length values
+
+        index_ =
+            clamp 0 length index
```
Comment (jonboiser) [inline src/Array/Extra.elm]: What is `clamp`? It doesn't seem to be in  scope in this file.

Comment (zwilias) [reply src/Array/Extra.elm]: https://package.elm-lang.org/packages/elm/core/latest/Basics#clamp

**Comment by zwilias** [comment]
I think there needs to be a tiny discussion on the expected behaviour with out of bound indexes.

There are 3 options:
- adopt the `Array.slice` behaviour, so `insertAt -1` will insert at the penultimate position in the array
- adopt the `Array.set` behaviour where the array isn't modified if the index is out of bounds
- clamp on both sides (which the current PR implements)

Of those three, the current choice seems furthest removed from how other functions in `Array` work. I'd suggest adopting one of the other two options, both of which are fairly easy to implement (removing the `clamp` line, or adding an `if/else` expression).

CC @prikhi as maintainer

**Comment by MartinSStewart** [comment]
I personally think option 3 is the most intuitive.  In light of it not being consistent with the rest of the array library I think option 2 is a good alternative (option 1 seems really unintuitive to me in the context of `insertAt`).

**Comment by MartinSStewart** [comment]
> 
> 
> Need to add the function to the `@docs` list in the module's documentation comment as well.

Fixed

**Comment by MartinSStewart** [comment]
Hi, I'm not sure who is responsible for it but could someone release a new version of array-extra with insertAt included?

**Comment by prikhi** [comment]
@jonboiser Is the actual maintainer, I just push releases when they're busy or not around.

**Pull request merged**